frontend/src/services/FakeDataSchemaService.ts

```diff
@@ -0,0 +1,123 @@
+import { injectable } from 'inversify';
+import 'reflect-metadata';
+import { EdgeType } from '../shared/schema/EdgeType';
+import { EntityType } from '../shared/schema/EntityType';
+import { EntityTypeAttribute } from '../shared/schema/EntityTypeAttribute';
+import { NodeType } from '../shared/schema/NodeType';
+import { CancellationToken } from '../utils/CancellationToken';
+import delay from '../utils/delay';
+import nouns from '../utils/nouns';
+import withCancellation from '../utils/withCancellation';
+import getRandomIndex from '../utils/getRandomIndex';
+import SchemaService from './SchemaService';
+
+// TODO: What else is possible?
+export const possibleAttributeTypes: string[] = [
+  'String',
+  'Long',
+  'StringArray',
+];
+
+@injectable()
+export class FakeDataSchemaService extends SchemaService {
+  private readonly usedNounIndices: Set<number> = new Set<number>();
+
+  private edgeTypesPromise: Promise<EdgeType[]> | null = null;
+
+  private nodeTypesPromise: Promise<NodeType[]> | null = null;
+
+  private getRandomNoun(): string {
+    let nounIndex = getRandomIndex(nouns.length);
+
+    while (this.usedNounIndices.has(nounIndex)) {
+      nounIndex = getRandomIndex(nouns.length);
+    }
+
+    this.usedNounIndices.add(nounIndex);
+    return nouns[nounIndex];
+  }
+
+  private generateRandomEntityTypeAttribute(): EntityTypeAttribute {
+    const name = this.getRandomNoun();
+
+    // Build the string array that contains the types the attribute may contain.
```

Comment:
maybe write a comment here "build a random subset of possibleAttributeTypes" as this took me some minutes to get at first